vizro-core/docs/pages/user-guides/dynamic-data.md

```diff
@@ -0,0 +1,133 @@
+            ],
+            controls=[vm.Filter(column="species")],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+        ```
+
+    === "Result"
+        [![DataBasic]][DataBasic]
+
+    [DataBasic]: ../../assets/user_guides/data/data_pandas_dataframe.png
+
+
+By default, when caching is turned on, dynamic data is cached in the Data Manager for 5 minutes. A refresh of the dashboard within this time interval will fetch the pandas DataFrame from the cache and not re-run the data loading function. Once the cache timeout period has elapsed, the next refresh of the dashboard will re-execute the dynamic data loading function. The resulting pandas DataFrame will again be put into the cache and not expire until another 5 minutes has elapsed.
+
+If you would like to alter some options, such as the default cache timeout, then you can specify a different cache configuration:
+
+```py title="Simple cache with timeout set to 10 minutes"
+data_manager.cache = Cache(config={"CACHE_TYPE": "SimpleCache", "CACHE_DEFAULT_TIMEOUT": 600})
+```
+
+!!! warning
+
+    Simple cache exists purely for single-process development purposes and is not intended to be used in production. If you deploy with multiple workers, [for example with gunicorn](run.md/#gunicorn), then you should use a production-ready cache backend. All of Flask-Caching's [built-in backends](https://flask-caching.readthedocs.io/en/latest/#built-in-cache-backends) other than `SimpleCache` are suitable for production. In particular, you might like to use [`FileSystemCache`](https://cachelib.readthedocs.io/en/stable/file/) or [`RedisCache`](https://cachelib.readthedocs.io/en/stable/redis/):
+
+    ```py title="Production-ready caches"
+    # Store cached data in CACHE_DIR
+    data_manager.cache = Cache(config={"CACHE_TYPE": "FileSystemCache", "CACHE_DIR": "cache"})
+
+    # Use Redis key-value store
+    data_manager.cache = Cache(config={"CACHE_TYPE": "RedisCache", "CACHE_REDIS_HOST": "localhost", "CACHE_REDIS_PORT": 6379})
+    ```
+
+Note that when a production-ready cache backend is used, the cache is persisted beyond the Vizro process and is not cleared by restarting your server. If you wish to clear the cache then you must do so manually, e.g. if you use `FileSystemCache` then you would delete your `cache` directory. Persisting the cache can also be useful for development purposes when handling data that takes a long time to load: even if you do not need the data to refresh while your dashboard is running, it can speed up your development loop to use dynamic data with a cache that is persisted between repeated runs of Vizro.
+
+### Configure timeouts
+
+You can change the timeout of the cache independently for each dynamic data source in the Data Manager using the `timeout` setting (measured in seconds). A `timeout` of 0 indicates that the cache does not expire. This is effectively the same as using [static data](static-data.md).
```

Comment:
I thought before about adding an option for "always expires" `timeout` data and decided not to bother. For now the solution is just to set a low timeout (the lowest possible is `timeout = 1`). Given that we have to handle network requests, which takes a while, and don't really deal with live-streamed data pushed through websocket or similar anyway, I think 1 second timeout basically counts good enough as "always expires" for our purposes.

(To me, ideally `timeout=0` sound like it should mean "always expires", and `timeout=inf` would mean "never expires". I considered implementing this but decided changing the meaning of `timeout=0` from what flask-caching does was too confusing, even though I think they've got the meaning of `timeout=0` the wrong way round...)

Does that make sense?

Reply:
I also thought about `timeout=0 # always expires` and `timeout=None # (or False) - never expires` option and it perfectly makes sense. However, I don't know how wise it is to change the behaviour for `timeout` comparing on how it behaves in the flask_cacking lib. 

Maybe we should stick with `timeout=0 # never expires` and then investigate and implement the option for `always expires` afterwards in a separate PR. Otherwise we need to implement it right now or to break it in the future?

Reply:
Yeah, I had exactly the same thoughts as you and decided it's also kind of confusing to have `timeout = 0` meaning the opposite of `timeout = False`, given that `False` is almost the same as `0` 😬 So probably `timeout = data_manager.ALWAYS` where `ALWAYS` is some sentinel value would be best here to be unambiguous.

Let's leave with the flask-caching convention that `timeout = 0` means "never expires" and then see whether people actually want an "alway expires" setting or if just setting a low timeout like `timeout = 1` is sufficient for user needs. If it's not then in future we can add `timeout = ALWAYS` without breaking anything.